### ISSUE TITLE
windows line ending instead of unix

### ISSUE THREAD

**Issue by oe1tkt** [widgetfactory/jce#13]
the released files 2.6.6 (via web update update from 2.6.2)
administrator/components/com_jce/controller/updates.php
administrator/components/com_jce/controller/popup.php
administrator/components/com_jce/controller/mediabox.php
administrator/components/com_jce/controller/help.php
administrator/components/com_jce/controller/cpanel.php
administrator/components/com_jce/controller/config.php
administrator/components/com_jce/jce.xml

 have CR/LF endings instead of LF (unix) .

**Comment by oe1tkt** [comment]
On my own joomla installation process, I automated the installation process to get rid of the CR/LF line endings and get information about wrong line endings.
The provided pkg_jce_268.zip via https://www.joomlacontenteditor.net/index.php?option=com_updates&view=update&format=xml&file=pkg_jce.xml
contains:

non-Unix line ending found in administrator/components/com_jce/controller/config.php
non-Unix line ending found in administrator/components/com_jce/controller/cpanel.php
non-Unix line ending found in administrator/components/com_jce/controller/help.php
non-Unix line ending found in administrator/components/com_jce/controller/mediabox.php
non-Unix line ending found in administrator/components/com_jce/controller/popup.php
non-Unix line ending found in administrator/components/com_jce/controller/updates.php
non-Unix line ending found in administrator/components/com_jce/jce.xml
non-Unix line ending found in administrator/components/com_jce/legacy.xml
non-Unix line ending found in administrator/components/com_jce/models/config.php
non-Unix line ending found in administrator/components/com_jce/models/config.xml
non-Unix line ending found in administrator/components/com_jce/models/cpanel.xml
non-Unix line ending found in administrator/components/com_jce/models/help.xml
non-Unix line ending found in administrator/components/com_jce/models/plugins.json
non-Unix line ending found in administrator/components/com_jce/models/plugins.php
non-Unix line ending found in administrator/components/com_jce/models/updates.php
non-Unix line ending found in administrator/components/com_jce/tables/profiles.php
non-Unix line ending found in administrator/components/com_jce/views/config/view.html.php
non-Unix line ending found in administrator/components/com_jce/views/cpanel/tmpl/default.php
non-Unix line ending found in administrator/components/com_jce/views/cpanel/view.html.php
non-Unix line ending found in administrator/components/com_jce/views/help/view.html.php
non-Unix line ending found in administrator/components/com_jce/views/mediabox/tmpl/default.php
non-Unix line ending found in administrator/components/com_jce/views/preferences/tmpl/default.php
non-Unix line ending found in administrator/components/com_jce/views/preferences/view.html.php
non-Unix line ending found in administrator/components/com_jce/views/profiles/tmpl/default.php
non-Unix line ending found in administrator/components/com_jce/views/profiles/tmpl/form.php
non-Unix line ending found in administrator/components/com_jce/views/updates/tmpl/default.php
non-Unix line ending found in administrator/components/com_jce/views/updates/view.html.php
non-Unix line ending found in administrator/components/com_jce/views/users/tmpl/default.php
non-Unix line ending found in administrator/language/en-GB/en-GB.com_jce.ini
non-Unix line ending found in administrator/language/en-GB/en-GB.com_jce.sys.ini
non-Unix line ending found in components/com_jce/editor/extensions/aggregator/dailymotion.xml
non-Unix line ending found in components/com_jce/editor/extensions/aggregator/vimeo.xml
non-Unix line ending found in components/com_jce/editor/extensions/aggregator/vine.xml
non-Unix line ending found in components/com_jce/editor/extensions/aggregator/youtube.xml
non-Unix line ending found in components/com_jce/editor/extensions/links/joomlalinks.php
non-Unix line ending found in components/com_jce/editor/extensions/links/joomlalinks/contact.php
non-Unix line ending found in components/com_jce/editor/extensions/links/joomlalinks.xml
non-Unix line ending found in components/com_jce/editor/extensions/links/joomlalinks/weblinks.php
non-Unix line ending found in components/com_jce/editor/libraries/classes/mobile.php
non-Unix line ending found in components/com_jce/editor/libraries/xml/config/editor.xml
non-Unix line ending found in components/com_jce/editor/libraries/xml/help/manager.xml
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/article/classes/config.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/browser/browser.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/browser/classes/config.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/charmap/classes/charmap.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/cleanup/classes/config.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/clipboard/classes/clipboard.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/clipboard/clipboard.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/colorpicker/classes/config.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/emotions/classes/config.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/imgmanager/classes/imgmanager.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/imgmanager/imgmanager.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/imgmanager/imgmanager.xml
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/searchreplace/classes/searchreplace.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/searchreplace/searchreplace.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/spellchecker/classes/pspell.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/spellchecker/classes/enchantspell.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/spellchecker/classes/spellchecker.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/style/classes/style.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/tabfocus/classes/config.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/table/classes/table.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/table/table.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/xhtmlxtras/classes/xhtmlxtras.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/plugins/xhtmlxtras/xhtmlxtras.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/themes/advanced/tmpl/colorpicker.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/themes/advanced/tmpl/shortcuts.php
non-Unix line ending found in components/com_jce/editor/tiny_mce/themes/advanced/tmpl/charmap.php
non-Unix line ending found in components/com_jce/jce.php
non-Unix line ending found in components/com_jce/views/popup/tmpl/default.php
non-Unix line ending found in components/com_jce/views/popup/view.html.php
non-Unix line ending found in jce.xml
non-Unix line ending found in administrator/language/en-GB/en-GB.plg_editors_jce.ini
non-Unix line ending found in administrator/language/en-GB/en-GB.plg_editors_jce.sys.ini
non-Unix line ending found in plugins/editors/jce/jce.xml
non-Unix line ending found in plugins/editors/jce/legacy.jce.xml
non-Unix line ending found in administrator/language/en-GB/en-GB.plg_content_jce.ini
non-Unix line ending found in administrator/language/en-GB/en-GB.plg_content_jce.sys.ini
non-Unix line ending found in administrator/language/en-GB/en-GB.plg_extension_jce.ini
non-Unix line ending found in administrator/language/en-GB/en-GB.plg_extension_jce.sys.ini
non-Unix line ending found in administrator/language/en-GB/en-GB.plg_system_jce.ini
non-Unix line ending found in administrator/language/en-GB/en-GB.plg_system_jce.sys.ini
Please update your zipping process for Joomla packages according to Joomla standard, using Unix line endings.

**Comment by ryandemmer** [comment]
Does the same package from - https://www.joomlacontenteditor.net/downloads/editor/core/item/jce-268 - include the CR/LF line endings?

**Comment by oe1tkt** [comment]
yes, I checked it again. If you unzip the whole package, you get some other zip files. For instance, inside directory "packages", there is a "com_jce.zip", unpacking this zip, you get the file "/administrator/components/com_jce/jce.xml" which is an CR/LF (DOS) file ending XML-file with date February 1st, 2017. The list above is the result after the installation process on Joomla. Normally, no one is in trouble to keep mixed DOS and Unix line endings, unfortunately we need strict Unix format for version control. In the past, I manually changed the files to Unix format. See also https://github.com/joomla/coding-standards/blob/master/manual/basic-guidelines.md
